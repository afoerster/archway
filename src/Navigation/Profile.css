.Profile {
    display: flex;
    flex-direction: row;
    color: white;
    padding: 7px;
    font-weight: 200;
    font-size: 28px;
    font-family: "Playfair Display";
}

.Profile-logout button {
    cursor: pointer;
    color: white;
    font-size: 14px;
    margin-left: 5px;
    font-family: "Roboto";
    font-weight: 200;
    text-decoration: none;
    text-transform: uppercase;
    background: none;
    border: none;
}