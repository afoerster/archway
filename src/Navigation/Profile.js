import React from 'react';
import {connect} from "react-redux";
import { Button } from 'antd';

import {requestLogout} from "../Auth/actions";

const Profile = ({profile, requestLogout}) =>  {
    let name;
    if (profile && profile.name)
        name = profile.name;

    return (
        <div style={{ width: '100%', padding: 15, color: 'white', textAlign: 'center', backgroundColor: '#415161' }}>
            <div>
                Hey, {name}
            </div>
            <Button ghost size="small" onClick={requestLogout}>
                Log Out
            </Button>
        </div>
    );
};

export default connect(
    state => state.auth,
    {requestLogout}
)(Profile);
