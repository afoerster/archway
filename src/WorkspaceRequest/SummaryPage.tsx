import * as React from 'react';
import { Card, Icon, Row, Col, List } from 'antd';
import { Workspace, Database } from '../WorkspaceListing/Workspace';

interface Props {
  workspace: Workspace
  name: string
}

const DatabaseItem = (name: string) => <h4>{name}</h4>;

const SummaryPage = ({ workspace, name }: Props) => {
  const databaseNames = workspace.data.map((database: Database) => database.name)
  return (
    <div>
      <h3>
        Here's a preview of what we'll request. Take a look and submit your request when you're ready!
    </h3>
      <Row type="flex" justify="center" style={{ marginTop: 25, marginBottom: 25 }}>
        <Col span={12}>
          <Card>
            <Icon type="crown" style={{ fontSize: 42 }} />
            <h3>You will be the liason for this project</h3>
            <h4>{name}</h4>
          </Card>
        </Col>
      </Row>
      <Row type="flex" justify="center" gutter={25} style={{ marginTop: 25, marginBottom: 25 }}>
        <Col span={12}>
          <Card>
            <Icon type="database" style={{ fontSize: 42 }} />
            <h3>The following databases will be created for you</h3>
            <List
              renderItem={DatabaseItem}
              dataSource={databaseNames} />
          </Card>
          <Card>
            <Icon type="thunderbolt" style={{ fontSize: 42 }} />
            <h3>You will be the liason for this project</h3>
            <h4>{name}</h4>
          </Card>
        </Col>
      </Row>
    </div>
  );
}

export default SummaryPage;