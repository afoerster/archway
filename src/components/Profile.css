.Profile {
    border-radius: 50%;
    font-size: 32px;
    color: black;
    padding: 7px;
    background-color: white;
    font-weight: 300;
}