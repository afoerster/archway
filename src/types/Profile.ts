export interface Profile {
  name: string;
  username: string;
  permissions: {
    risk_management: boolean
    platform_operations: boolean,
  };
}
