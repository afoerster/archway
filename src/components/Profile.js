import React from 'react';
import "./Profile.css";
import {connect} from "react-redux";

const Profile = ({profile}) =>  {
    return (
        <div className="Profile">
            {profile.initials}
        </div>
    );
};

export default connect(
    state => state.account,
    {}
)(Profile);