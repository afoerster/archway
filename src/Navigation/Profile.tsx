import { Avatar } from 'antd';
import * as React from 'react';
import { connect } from "react-redux";
import { Dispatch } from 'redux';
import { createStructuredSelector } from 'reselect';
import { requestLogout } from "../Auth/actions";
import { Profile } from '../types/Profile';
import { getProfile } from '../selectors';


interface Props {
  profile: Profile,
  doLogout: () => void
}

const Profile = ({profile, doLogout}: Props) =>  {
    let name;
    if (profile && profile.name)
        name = profile.name;

    return (
        <div style={{ width: '100%', padding: 15, color: 'white', textAlign: 'center', borderTop: '1px solid #C6CACF', borderBottom: '1px solid #C6CACF', backgroundColor: '#415161' }}>
            <Avatar shape="circle" size="large" icon="user" style={{ backgroundColor: 'transparent', color: 'white' }} />
            <h3 style={{ color: 'white' }}>
                hey, {name}!
            </h3>
            <h6>
              <a href="#" style={{ textTransform: 'uppercase', fontWeight: 200, color: 'white' }} onClick={doLogout}>
                  log out
              </a>
            </h6>
        </div>
    );
};

const mapStateToProps = () =>
  createStructuredSelector({
    profile: getProfile(),
  });

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  doLogout: () => dispatch(requestLogout())
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
