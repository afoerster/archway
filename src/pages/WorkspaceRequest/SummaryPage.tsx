import { Card, Col, Icon, List, Row } from 'antd';
import * as React from 'react';
import { Database, ResourcePool, Workspace } from '../../types/Workspace';

interface Props {
  workspace?: Workspace;
  name: string;
}

const ResourceItem = (name: string) => <h4>{name}</h4>;

const SummaryPage = ({ workspace, name }: Props) => {
  const databaseNames = workspace && workspace.data.map((database: Database) => database.name);
  const poolNames = workspace && workspace.processing.map((pool: ResourcePool) => pool.pool_name);
  return (
    <div>
      <h3>
        Here's a preview of what we'll request. Take a look and submit your request when you're ready!
    </h3>
      <Row type="flex" justify="center" style={{ marginTop: 25, marginBottom: 25 }}>
        <Col span={8}>
          <Card>
            <Icon type="crown" style={{ fontSize: 42 }} />
            <h3>You will be the liason for this project</h3>
            <h4>{name}</h4>
          </Card>
        </Col>
      </Row>
      <Row type="flex" justify="center" gutter={25} style={{ marginTop: 25, marginBottom: 25 }}>
        <Col span={8}>
          <Card>
            <Icon type="database" style={{ fontSize: 42 }} />
            <h3>The following <b>hive databases</b> will be created for you</h3>
            <List
              renderItem={ResourceItem}
              dataSource={databaseNames} />
          </Card>
        </Col>
        <Col span={8}>
          <Card>
            <Icon type="thunderbolt" style={{ fontSize: 42 }} />
            <h3>The following <b>resource pools</b> will be created for you</h3>
            <List
              renderItem={ResourceItem}
              dataSource={poolNames} />
          </Card>
        </Col>
      </Row>
    </div>
  );
};

export default SummaryPage;
