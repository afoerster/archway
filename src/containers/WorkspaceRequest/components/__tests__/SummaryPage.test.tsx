import * as React from 'react';
import { shallow } from 'enzyme';

import SummaryPage from '../SummaryPage';
import { Profile } from '../../../../models/Profile';

describe('SummaryPage', () => {
  it('renders correctly', () => {
    const profile: Profile = {
      name: '',
      username: '',
      permissions: {
        risk_management: false,
        platform_operations: false,
      },
    };
    const wrapper = shallow(<SummaryPage profile={profile} />);
    expect(wrapper).toMatchSnapshot();
  });
});
