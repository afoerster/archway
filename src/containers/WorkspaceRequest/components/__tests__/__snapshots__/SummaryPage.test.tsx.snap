// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryPage renders correctly 1`] = `ShallowWrapper {}`;
